src/probnum/statespace/sde.py

```diff
@@ -298,6 +353,45 @@ def f(t, y):
 
         return f, y0
 
+    def _setup_vectorized_mde_forward_sqrt(self, initrv, _diffusion=1.0):
+        """Set up forward moment differential equations (MDEs) using a square-root
+        implementation. (https://ieeexplore.ieee.org/document/4045974)
+
+        Compute an ODE vector field that represents the MDEs and is
+        compatible with scipy.solve_ivp.
+        """
+        dim = len(initrv.mean)
+
+        def f(t, y):
+            # Undo vectorization
+            mean, cov_sqrt_flat = y[:dim], y[dim:]
+            cov_sqrt = cov_sqrt_flat.reshape((dim, dim))
+
+            # Apply iteration
+            F = self.driftmatfun(t)
+            u = self.forcevecfun(t)
+            L = self.dispmatfun(t)
+
+            new_mean = F @ mean + u
+            cov_sqrt_lu, cov_sqrt_piv = scipy.linalg.lu_factor(cov_sqrt)
+            F_bar = scipy.linalg.lu_solve((cov_sqrt_lu, cov_sqrt_piv), F @ cov_sqrt)
+            L_bar = np.sqrt(_diffusion) * scipy.linalg.lu_solve(
+                (cov_sqrt_lu, cov_sqrt_piv), L
+            )
+            M = F_bar + F_bar.T + L_bar @ L_bar.T
+
+            new_cov_sqrt = cov_sqrt @ (np.tril(M, -1) + 1 / 2 * np.diag(np.diag(M)))
+
+            # Vectorize outcome
+            new_cov_sqrt_flat = new_cov_sqrt.flatten()
+            y_new = np.hstack((new_mean, new_cov_sqrt_flat))
+            return y_new
+
+        initcov_sqrt_flat = initrv.cov_cholesky.flatten()
+        y0 = np.hstack((initrv.mean, initcov_sqrt_flat))
+
+        return f, y0
```

Comment:
Do you think it would be possible to document this a bit more extensively than linking the paper? It is paywall-protected, so not everyone can see the equation easily. One way would be to write down the actual equation (in a maths way) that is solved here. That would help me when reading the code :)

Reply:
I can do that.